### PR TITLE
[MIG] School Attendance: Migrated school Attendance into 16.0 and sol…

### PR DESCRIPTION
**Pull request by RiteshSerpentCS**
…ved pre-commits

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@nikul-serpentcs pls review the attendance and see if we can remove some extra code

File: school_attendance/wizard/attendance_sheet_wizard.py [hunk 1]
```diff
@@ -9,33 +9,27 @@ class MonthlyAttendanceSheet(models.TransientModel):
...
 
     def monthly_attendance_sheet_open_window(self):
-        """ This method open new window with monthly attendance sheet
-            @param self : Object Pointer
-            @param cr : Database Cursor
-            @param uid : Current Logged in User
-            @param ids : Current Records
-            @param context : standard Dictionary
-            @return : record of monthly attendance sheet
+        """This method open new window with monthly attendance sheet
```
Comment (nikul-serpentcs) [inline school_attendance/wizard/attendance_sheet_wizard.py:17]: Remove plz

File: school_attendance/views/school_attendance_view.xml [hunk 2]
```diff
@@ -137,62 +146,116 @@
...
-                            <field name="student_id" domain="[('state','=','done')]"
-                                   options="{&quot;no_create&quot;: True}"
-                                   attrs="{'readonly':[('state','in',['approve','reject','toapprove'])]}"
-                                   context="{'student_id':True}"/>
-                            <field name="roll_no" readonly="1"/>
-                            <field name="standard_id" readonly="1"/>
+                            <field
+                                name="student_id"
+                                domain="[('state','=','done')]"
+                                options="{&quot;no_create&quot;: True}"
```
Comment (nikul-serpentcs) [inline school_attendance/views/school_attendance_view.xml:224]:
```suggestion
                                options="{'no_create': True}"
```

File: school_attendance/wizard/monthly_attendance_wizard_view.xml [hunk 3]
```diff
@@ -9,25 +9,44 @@
...
                     <group colspan="4">
                         <group>
-                            <field name="academic_year_id" required="1"
-                                 options="{&quot;no_create&quot;: True, &quot;no_search&quot;: True, &quot;no_open&quot;: True}"/>
-                            <field name="month" required="1"/>
-                            <field name="month_str" required="1" invisible="1"/>
+                            <field
+                                name="academic_year_id"
+                                required="1"
+                                options="{&quot;no_create&quot;: True, &quot;no_search&quot;: True, &quot;no_open&quot;: True}"
```
Comment (nikul-serpentcs) [inline school_attendance/wizard/monthly_attendance_wizard_view.xml:15]: **no_search**  This is the first time I have encountered this attribute in my experience with Odoo. :red_circle:

File: school_attendance/wizard/monthly_attendance_wizard_view.xml [hunk 4]
```diff
@@ -9,25 +9,44 @@
...
                         <group>
-                            <field name="course_id" required="1"
-                                options="{&quot;no_create&quot;: True, &quot;no_search&quot;: True, &quot;no_open&quot;: True}"/>
-                            <field name="is_elective_subject"/>
-                            <field name="subject_id" attrs="{'invisible':[('is_elective_subject','=', False)], 'required':[('is_elective_subject','!=', False)]}"
-                                options="{&quot;no_create&quot;: True}" domain="[('is_elective_subject', '=', True)]"/>
+                            <field
+                                name="course_id"
+                                required="1"
+                                options="{&quot;no_create&quot;: True, &quot;no_search&quot;: True, &quot;no_open&quot;: True}"
```
Comment (nikul-serpentcs) [inline school_attendance/wizard/monthly_attendance_wizard_view.xml:24]: same here too.

File: school_attendance/wizard/student_attendance_by_month.py [hunk 5]
```diff
@@ -33,7 +31,7 @@ def default_get(self, fields):
         return res
 
     def print_report(self):
-        """ This method prints report
+        """This method prints report
         @param self : Object Pointer
         @param cr : Database Cursor
```
Comment (nikul-serpentcs) [inline school_attendance/wizard/student_attendance_by_month.py:36]: Remove Please

File: school_attendance/wizard/monthly_attendance_wizard_view.xml [hunk 6]
```diff
@@ -41,9 +60,13 @@
...
-        parent="school_attendance.menu_attendance" action="report_monthly"
-        sequence="51" />
+    <menuitem
+        id="menu_monthly_attendances"
+        name="Monthly Attendance Report"
+        parent="school_attendance.menu_attendance"
+        action="report_monthly"
+        sequence="51"
+    />
         <!-- groups="school.group_school_administration" -->
```
Comment (nikul-serpentcs) [inline school_attendance/wizard/monthly_attendance_wizard_view.xml:70]: remove dead code

File: school_attendance/models/school_attendance.py [hunk 7]
```diff
@@ -252,9 +247,7 @@ def check_student_request(self):
     @api.constrains("start_date", "end_date")
     def check_dates(self):
         if self.start_date > self.end_date:
```
Comment (nikul-serpentcs) [inline school_attendance/models/school_attendance.py:249]:
```suggestion
        if self.start_date and self.end_date and self.start_date > self.end_date:
```

File: school_attendance/wizard/monthly_attendance_wizard.py [hunk 8]
```diff
@@ -27,16 +29,12 @@ def _get_current_academic_year(self):
         "academic.year",
         ondelete="restrict",
         string="Academic Session",
-        default=lambda obj: obj.env["academic.year"].search(
-            [("current", "=", True)]
-        ),
+        default=lambda obj: obj.env["academic.year"].search([("current", "=", True)]),
```
Comment (nikul-serpentcs) [inline school_attendance/wizard/monthly_attendance_wizard.py:32]:
```suggestion
        default=lambda obj: obj.env["academic.year"].search([("current", "=", True)], limit=1),
```

**Review by nikul-serpentcs — COMMENTED** [review]
Quick review

**Pull request merged**